internal/timeutil/time_bench_test.go

```diff
@@ -0,0 +1,189 @@
+package timeutil
+
+import (
+	"testing"
+	"time"
+)
+
+func BenchmarkParse(b *testing.B) {
+	type args struct {
+		t string
+	}
+	type test struct {
+		name string
+		args args
+	}
+	tests := []test{
+		{
+			name: "t 1ns",
+			args: args{
+				t: "1ns",
+			},
+		},
+		{
+			name: "t 1µs",
+			args: args{
+				t: "1µs",
+			},
+		},
+		{
+			name: "t 1ms",
+			args: args{
+				t: "1ms",
+			},
+		},
+		{
+			name: "t 1s",
+			args: args{
+				t: "1s",
+			},
+		},
+		{
+			name: "t 1m",
+			args: args{
+				t: "1m",
+			},
+		},
+		{
+			name: "t 1h",
+			args: args{
+				t: "1h",
+			},
+		},
+	}
+	for _, test := range tests {
+		b.Run(test.name, func(b *testing.B) {
+			b.RunParallel(func(pb *testing.PB) {
+				for pb.Next() {
+					got, err := Parse(test.args.t)
+					if err != nil {
+						b.Error(err)
+						return
+					}
+					if got == 0 {
+						b.Error("got is 0")
+					}
+				}
+			})
+		})
+	}
+}
+
+func BenchmarkParseWithDefault(b *testing.B) {
+	type args struct {
+		t string
+		d time.Duration
+	}
+	type test struct {
+		name string
+		args args
+	}
+	tests := []test{
+		{
+			name: "t 1ns and d 50 millisecond",
+			args: args{
+				t: "1ns",
+				d: 50 * time.Millisecond,
+			},
+		},
+		{
+			name: "t 1µs and d 50 millisecond",
+			args: args{
+				t: "1µs",
+				d: 50 * time.Millisecond,
+			},
+		},
+		{
+			name: "t 1ms and d 50 millisecond",
+			args: args{
+				t: "1ms",
+				d: 50 * time.Millisecond,
+			},
+		},
+		{
+			name: "t 1s and d 50 millisecond",
+			args: args{
+				t: "1s",
+				d: 50 * time.Millisecond,
+			},
+		},
+		{
+			name: "t 1m and d 50 millisecond",
+			args: args{
+				t: "1m",
+				d: 50 * time.Millisecond,
+			},
+		},
+		{
+			name: "t 1h and d 50 millisecond",
+			args: args{
+				t: "1h",
+				d: 50 * time.Millisecond,
+			},
+		},
+
+		// The following are error pattern.
```

Comment:
Why do we need error pattern here?

Reply:
If an error occurs, the default value will be set, so take that into consideration.

Reply:
btw
I've added the following error pattern cases to take into account the length of the string when an error occurs.